rclcpp/include/rclcpp/executor_options.hpp

```diff
@@ -0,0 +1,49 @@
+// Copyright 2014 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__EXECUTOR_OPTIONS_HPP_
+#define RCLCPP__EXECUTOR_OPTIONS_HPP_
+
+#include "rclcpp/context.hpp"
+#include "rclcpp/contexts/default_context.hpp"
+#include "rclcpp/memory_strategies.hpp"
+#include "rclcpp/memory_strategy.hpp"
+#include "rclcpp/visibility_control.hpp"
+
+namespace rclcpp
+{
+
+/// Options to be passed to the executor constructor.
+struct ExecutorOptions
+{
+  ExecutorOptions()
+  : memory_strategy(rclcpp::memory_strategies::create_default_strategy()),
+    context(rclcpp::contexts::get_global_default_context()),
+    max_conditions(0)
+  {}
+
+  rclcpp::memory_strategy::MemoryStrategy::SharedPtr memory_strategy;
+  rclcpp::Context::SharedPtr context;
+  size_t max_conditions;
+};
+
+namespace executor
+{
+
+using ExecutorArgs [[deprecated("use rclcpp::ExecutorOptions instead")]] = ExecutorOptions;
+
+}  // namespace executor
```

Comment:
Need a deprecation for `rclcpp::executor::create_default_executor_arguments()`?

Reply:
I forgot to mention this in the issue description, thanks for pointing it out.

I ended up just removing it (not moving it) because there's no reason not to just use the `ExecutorOptions` default constructor instead. It also caused the header to not compile without warnings because it was a static function that had no uses in the header.

I audited our code at least, and didn't find it being used in many places, so I would like to just remove it, though I can add a deprecation for it if you like.

Reply:
I agree it's pointless and probably unused. I would recommend adding a deprecation just for the sake of saying we followed this process consistently.

Reply:
Ok, will do.

Reply:
0548061